src/Workspaces/Core/Portable/Workspace/Host/TemporaryStorage/ITemporaryStreamStorageExtensions.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+
+namespace Microsoft.CodeAnalysis.Host
+{
+    internal static class ITemporaryStreamStorageExtensions
+    {
+        public static void WriteString(this ITemporaryStreamStorage storage, string value)
+        {
+            using var memoryStream = new MemoryStream();
```

Comment:
Can we use the `MemoryStream(byte[])` constructor to avoid having to allocate our own `StreamWriter`? 

`new MemoryStream(Encoding.UTF8Encoding.GetBytes(value))` 

Not sure how much that saves us, but it seems desirable to allocate the correct size stream if we know it.

Reply:
i switched to a pooled approach so we can avoid some intermediary garbage.  note, we still need the StreamWriter, but that's a tiny object (compared the byte[]s we save with pooling).